src/ao-profile/registry.lua

```diff
@@ -57,18 +49,141 @@ local function process_profile_action(msg, profile_id_to_check_for_update)
         })
         return
     end
+    
+    local tags = msg.Tags or {}
+
+    local upsert_metadata_stmt = [[
```

Comment:
I think We shouldn't bring this code back . If anything, use db select to see if we have the id, and use either create or update code below.  We can support v0.1 without using exactly the same code.